### PR TITLE
feat(server): centralized structured request logging (#261)

### PR DESCRIPTION
**Pull request by William-Hill**
## Summary
Closes #261 (follow-up to #30, task 5). Adds process-wide structured logging and a correlatable request id.

- **`server/lib/logging_config.py`** — root-logger JSON config (`configure_logging()`, idempotent), `LOG_LEVEL` env (default INFO), optional `LOG_FORMAT=plain` for local dev. `JsonLogFormatter` matches the existing `vision_tagger._log_call` line shape and merges `extra=` fields. `ContextFilter` stamps `request_id`/`user_id` from contextvars onto **every** record, so existing ad-hoc module logs become correlatable with zero call-site changes.
- **`server/lib/access_log.py`** — `AccessLogMiddleware`, **pure ASGI** (not `BaseHTTPMiddleware`, which copies the context and would sever the contextvar from the endpoint). One access line per request: `method, path, status, latency_ms, request_id, user_id`. Honors inbound `X-Request-ID`, echoes it on the response.
- **`server/services/auth.py`** — `current_auth` (async, runs in the main request task) sets `user_id_var`, so the access line and all module logs for the request carry the caller id.
- **`server/main.py`** — `configure_logging()` at import; `AccessLogMiddleware` added outermost.

### PII guard
Path only — never the query string (can carry tokens), body, or `Authorization` header. No client IP.

### Design note
Sync `def` routes run in a threadpool (copied context); the request-id propagation relies on the pure-ASGI middleware setting the contextvar before `await app()`, and `user_id` is set by the **async** `current_auth` dependency in the main task — both verified by tests.

## Test plan
- [x] `tests/test_logging.py` — 10 tests: formatter JSON validity + `extra` merge; `ContextFilter`; `configure_logging` idempotency + `LOG_LEVEL`; access line emitted once with core fields; response carries `X-Request-ID`; inbound id honored; **module log shares request id with access line** (core acceptance); query string not logged; `user_id` stamped via async dependency.
- [x] Full suite: 350 passed (340 baseline + 10 new), 0 regressions. The 9 errors in `test_tryon_migration_invariants.py` are pre-existing and unrelated (stale `00008_tryon.sql` path from the #208 migration rename).
- [ ] Manual: confirm Railway/Supabase logs show one JSON access line per request with correlatable `request_id`.

## Summary by CodeRabbit

* **New Features**
  * Structured newline-delimited JSON logging with per-request and per-user IDs on every log.
  * HTTP access logging: method, path (query removed), status, latency, and exactly one x-request-id on responses.
  * Safe handling of inbound request IDs (rejects malformed/oversized values and falls back to a generated ID); healthcheck paths excluded.
  * Idempotent logging configuration honoring format and level settings.

* **Tests**
  * Comprehensive unit and integration tests covering formatter output, context propagation, access logs, header behavior, security edge cases, and error logging.

[Review Change Stack](https://app.coderabbit.ai/change-stack/William-Hill/ptso_agent/pull/265?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: server/main.py [hunk 1]
```diff
@@ -23,6 +27,10 @@ async def lifespan(app: FastAPI):
 
 app = FastAPI(title="PTSO Agent API", version="0.1.0", lifespan=lifespan)
 
+# Added last so it wraps outermost: sets the request-id context before any
+# other middleware/route runs and logs the final status after they return.
+app.add_middleware(AccessLogMiddleware)
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/main.py]: Fixed in 4ddcb31 — AccessLogMiddleware is now registered after CORSMiddleware so Starlette wraps it outermost; a CORS-answered preflight OPTIONS now gets an access line + X-Request-ID. Regression test: test_access_middleware_wraps_outermost_including_cors.

File: server/main.py [hunk 2]
```diff
@@ -23,6 +27,10 @@ async def lifespan(app: FastAPI):
...
+
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/main.py]: Fixed in 4ddcb31 — moved app.add_middleware(AccessLogMiddleware) to after the CORS registration (added last → outermost), matching your suggestion. Covered by test_access_middleware_wraps_outermost_including_cors.

File: server/lib/logging_config.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    """Render a record as a single compact JSON object."""
+
+    def format(self, record: logging.LogRecord) -> str:
+        payload: dict[str, object] = {
+            "ts": self.formatTime(record, "%Y-%m-%dT%H:%M:%S%z"),
+            "level": record.levelname,
+            "logger": record.name,
+            "request_id": getattr(record, "request_id", None),
+            "user_id": getattr(record, "user_id", None),
+            "message": record.getMessage(),
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/lib/logging_config.py]: Fixed in 4ddcb31 — JsonLogFormatter now detects a JSON-object message and lifts its keys to top level (core fields protected), so vision_tagger._log_call keeps event/status/model as queryable fields instead of an escaped string. No call-site migration needed. Test: test_json_object_message_is_lifted_to_top_level.

File: server/lib/access_log.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        candidate = inbound.decode("latin-1") if inbound else ""
+        request_id = (
+            candidate if _REQUEST_ID_RE.match(candidate) else uuid.uuid4().hex
+        )
+
+        rid_token = request_id_var.set(request_id)
+        uid_token = user_id_var.set(None)
+        start = time.perf_counter()
+        status_code = 500
+        error: Exception | None = None
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/lib/access_log.py]: Fixed in 4ddcb31 — error is now BaseException and the handler catches BaseException, so asyncio.CancelledError (client disconnect/shutdown) is logged at ERROR with error=CancelledError and re-raised, not recorded as success. Test: test_cancelled_request_is_logged_as_error.

File: server/lib/access_log.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+        async def send_wrapper(message):
+            nonlocal status_code
+            if message["type"] == "http.response.start":
+                status_code = message["status"]
+                message.setdefault("headers", [])
+                message["headers"].append(
+                    (_REQUEST_ID_HEADER, request_id.encode("latin-1"))
+                )
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/lib/access_log.py]: Fixed in 4ddcb31 — send_wrapper now scans response headers and replaces an existing x-request-id in place, appending only when absent, so the client never sees a duplicate or a divergent value. Regression test: test_existing_request_id_header_is_replaced_not_duplicated.

File: tests/test_logging.py [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+    @app.get("/whoami")
+    def whoami(_=Depends(fake_auth)):
+        logging.getLogger("test.inner").info("did work")
+        return {}
+
+    TestClient(app).get("/whoami")
+    access = [r for r in capture.records if r.get("event") == "http_request"][0]
+    inner = [r for r in capture.records if r["logger"] == "test.inner"][0]
+    assert access["user_id"] == "user-77"
+    assert inner["user_id"] == "user-77"
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply tests/test_logging.py:192]: Investigated and the leak premise doesn't hold: TestClient runs the app in its own context, so the set() is confined to the request and never reaches other tests. Applying the suggested cross-context user_id_var.reset() actually raises (Token was created in a different Context). Documented this with a WHY comment in 4680e2f rather than adding a broken reset. Bundled Ruff nits (B008 Depends-in-default, RUF015 [0], ANN202) match repo-wide test conventions, intentionally left as-is.

File: tests/test_logging.py [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+    assert line["method"] == "GET"
+    assert line["path"] == "/ping"
+    assert line["status"] == 200
+    assert isinstance(line["latency_ms"], (int, float))
+
+
+def test_response_carries_request_id_header(capture):
+    client = TestClient(_app_that_logs())
+    resp = client.get("/ping")
+    assert resp.headers.get("x-request-id")
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply tests/test_logging.py]: Fixed in 0892c86 — applied the suggested tightening: the test now asserts the response x-request-id equals the logged access line's request_id, covering the no-inbound-ID correlation path (also makes the capture fixture used, clearing the Ruff ARG001 nit).
